### PR TITLE
Add kind/kep label

### PR DESCRIPTION
**Pull request by justaugustus**
/sig pm architecture

rel:
- https://github.com/kubernetes/community/pull/2655
- https://kubernetes.slack.com/archives/C09QZ4DQB/p1537137652000100

Signed-off-by: Stephen Augustus <foo@agst.us>

### CODE REVIEW

**Comment by cblecker** [comment]
/hold
Two things..
First, I think this might be a better fit for “area/“ rather than “kind/“. We just did a cleanup of kind/ (cc @jberkus @spiffxp)

Second, does every repo in all kubernetes orgs need this? Or is this more for just k/community and k/features?

**Comment by justaugustus** [comment]
@cblecker --
1. Seems that we primarily use `area/*` to signify areas of code. As KEPs are cross-cutting, I would hesitate to add them as a specific "area". We have some precedent for using `kind` in instances like this (`kind/feature`, `kind/design`) and KEPs would fall into a similar designation.
2. This would likely only be used for k/community and k/features. I forgot we had the per-repo labels now. Will fix that up.

**Comment by justaugustus** [comment]
Hey @k8s-ci-robot, promise I'm actually running the script. See?
```
[augustus@auggiebook01 test-infra]$ ./hack/update-labels.sh
INFO: Analysed target //label_sync:label_sync (0 packages loaded).
INFO: Found 1 target...
Target //label_sync:label_sync up-to-date:
  bazel-bin/label_sync/linux_amd64_stripped/label_sync
INFO: Elapsed time: 0.573s, Critical Path: 0.19s
INFO: 0 processes.
INFO: Build completed successfully, 1 total action
INFO: Running command line: bazel-bin/label_sync/linux_amd64_stripped/label_sync '--config=/home/augustus/prj/k8s/test-infra/label_sync/labels.yaml' '--action=docs' '--docs-template=/home/augustus/prj/k8s/test-infra/label_sync/labels.md.tmpl' '--docs-output=/home/augustus/prj/k8s/test-infra/label_sync/labels.md' '--css-template=/home/augustus/prj/k8s/test-infra/label_sync/labels.css.tmpl' '--css-output=/home/augustus/pINFO: Build completed successfully, 1 total action
WARN[0000] Repo isn't inside orgs                        org=kubernetes orgs= repo=kubernetes/community
WARN[0000] Repo isn't inside orgs                        org=kubernetes orgs= repo=kubernetes/features
WARN[0000] Repo isn't inside orgs                        org=kubernetes orgs= repo=kubernetes/kubernetes
WARN[0000] Repo isn't inside orgs                        org=kubernetes orgs= repo=kubernetes/test-infra
```

/test pull-test-infra-verify-labels

**Comment by justaugustus** [comment]
(Fun fact: the `pull-test-infra-verify-labels` job will fail if the label update and hack script aren't committed together. Dropped a PR to update the instructions: https://github.com/kubernetes/test-infra/pull/9439)

**Comment by fejta** [comment]
/uncc

I'm happy if cblecker is happy

**Comment by spiffxp** [comment]
As much as I cringe to see another `kind/` label, as long as it's restricted to those two repos I guess I'm OK with it.  To me area ~= subproject, and while I do think "make keps usable" is itself a subproject (ie: `area/keps`), I don't think an individual kep is part of that subproject.

Ideally this would be obviated by moving keps to their own repo, so you could just query for `repo:kubernetes/keps` instead of `org:kubernetes label:kind/kep`

File: label_sync/labels.yaml [hunk 1]
```diff
@@ -864,3 +864,19 @@ repos:
         name: area/prow/pod-utilities
         target: both
         addedBy: label
+  kubernetes/community:
+    labels:
+      - color: cc99ff
+        description: Categorizes issue or PR as related to a KEP.
```
Comment (spiffxp) [inline label_sync/labels.yaml]: That said, I don't see mention of the reason or need for `kind/kep` in the meta-kep proposal.  This description is too vague.  ie: what I do not want to see is PR's that implement a KEP being labeled as `kind/kep`, that is not a kind of work.

I was assuming it was solely to be able to query what PR's are KEPs as an interim to transitioning them into kubernetes/keps.  Which now that I think of it, does seem like an appropriate use for an `area/keps` label, to signify something is going to be done with these as part of that effort.

**Comment by spiffxp** [comment]
tl;dr I just talked myself back into wanting to block this.  Why do you need this label?

**Comment by BenTheElder** [comment]
What are your thoughts on keps having their own repo? @cblecker @justaugustus ? This sounds nice to me. IMO we should try more focused repos and less heavily overloaded ones. It would also help and with interested parties (reviewers?) trying to subscribe to KEPs without signing up to monitor _all_ community changes.

**Comment by dims** [comment]
+1 to a `kubernetes/keps` will be helpful to just monitor that.

**Comment by justaugustus** [comment]
@spiffxp -- agreed that `area/keps` would be akin to a KEP subproject, which I'm not sure that I / we want / need to bother spinning up a subproject for KEPs. I think `sig/pm` would suffice as the categorization and I've added a PR so that `keps/` gets properly categorized (https://github.com/kubernetes/community/pull/2677).

> Ideally this would be obviated by moving keps to their own repo, so you could just query for `repo:kubernetes/keps` instead of `org:kubernetes label:kind/kep`

You read my mind! 
Discussion about moving keps to their own repo starts [here](https://github.com/kubernetes/community/issues/2565#issuecomment-418021617), has been mentioned in the last few SIG Arch meetings, and is formalized in [KEP-1a](https://github.com/kubernetes/community/pull/2655).

In the interim, prior to everything getting moved over, `kind/kep` would be overloaded...

For `k/community`:
- Label incoming KEPs as `kind/kep`
- Enable `org:kubernetes label:kind/kep`, so we can identify active PRs to `k/community` and reroute the PR authors to `k/keps` (depending on the state)

For `k/keps` (fka `k/features`):
- Classify KEP submissions / tracking issues as `kind/kep`, differentiating them from `kind/feature`

Once everything is migrated to `k/keps`, we'd obviate the need for having the label in `k/community` and it would be removed.

@BenTheElder / @dims -- Yep. This was exactly the motivation for this change: https://github.com/kubernetes/community/issues/2565#issuecomment-414751095

Let me know if that's enough context to get this unblocked.

**Comment by BenTheElder** [comment]
Thanks for the detailed context! I had not seen all of this :-)
I'll leave the rest of the label wrangling to @spiffxp and @cblecker, but I'll be glad to see k/keps and will likely subscribe there as well 😄

**Comment by spiffxp** [comment]
@justaugustus So, this level of detail is great, but it should be in https://github.com/kubernetes/community/pull/2655.  Iterate on your plan and then we can execute.  

I also think the label descriptions need to be less vague in this case.  It sounds like your intent is for PRs to be labeled with `kind/kep` if they're touching a kep, and issues to be labeled with `kind/kep` if they're an umbrella/tracking issue for the implementation of a kep.

**Comment by justaugustus** [comment]
I definitely ran `hack/update-labels.sh` and squashed before force-pushing this again.
/test pull-test-infra-verify-labels

**Comment by spiffxp** [comment]
I'd like to see KEP-1a get to implementable before moving on this

**Comment by justaugustus** [comment]
Label CSS is fixed in https://github.com/kubernetes/test-infra/pull/9633

**Comment by justaugustus** [comment]
With KEP-1a merged, this is ready to go as well.
